PowerEditor/src/NppBigSwitch.cpp

```diff
@@ -1975,7 +1976,9 @@ LRESULT Notepad_plus::process(HWND hwnd, UINT message, WPARAM wParam, LPARAM lPa
 					scintillaContextmenu.create(hwnd, tmp, _mainMenuHandle, copyLink);
 					
 					POINT p;
-					if (lParam == -1)
+					int xPos = GET_X_LPARAM(lParam);
+					int yPos = GET_Y_LPARAM(lParam);
+					if ((xPos == -1) && (yPos == -1))

```

Comment:
I would do the following:
```cpp
					p.x = GET_X_LPARAM(lParam);
					p.y = GET_Y_LPARAM(lParam);
					if ((p.x== -1) && (p.y == -1))
					{
						// context menu activated via keyboard; pop up at text caret position
						...
					}
```
And remove
```cpp
					else
					{
						// context menu activated via mouse right-click; pop up at click location
						::GetCursorPos(&p);
					}
```

Does it make more sense?

Reply:
Sure, I wasn't sure how much change would be tolerated here; I had thought you liked `GetCursorPos` for some reason...